src/libsystemd-network/sd-ipv4ll.c

```diff
@@ -202,7 +200,7 @@ int sd_ipv4ll_set_address_seed(sd_ipv4ll *ll, unsigned seed) {
         if (!random_data_state)
                 return -ENOMEM;
 
-        r = initstate_r(seed, random_data_state, 128, random_data);
+        r = initstate_r((unsigned) seed, random_data_state, 128, random_data);
```

Comment:
Hm, doesn't this just push the problem around? If initstate_r takes an unsigned, which is shorter than uint64_t (at least on some arches), a certain value of seed specified for sd_ipv5ll_set_address_seed will yield different behaviours. So maybe it's better to bite the bullet and specify the size of the seed as the _minimum_ of available sizes, i.e. uint32_t, to have consistent interpretation of the argument on all architectures. After all 2^32 should be enough seeds.